Integrations/src/main/java/io/deephaven/integrations/learn/Future.java

```diff
@@ -0,0 +1,64 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.integrations.python.PythonFunction;
+import org.jpy.PyObject;
+
+/**
+ * Performs a deferred computation on a portion of a table.
+ */
+public class Future {
+
+    private final PyObject func;
+    private final Input[] inputs;
+    private final IndexSet indexSet;
+    private boolean called;
+    private PyObject result;
+
+    /**
+     * Creates a new Future.
+     *
+     * @param func Function that performs computation on gathered data.
+     * @param inputs Inputs to the Future computation.
+     * @param batchSize Maximum number of rows for deferred computation.
+     */
+    public Future(PyObject func, int batchSize, Input[] inputs) {
+
+        this.func = func;
+        this.inputs = inputs;
+        this.indexSet = new IndexSet(batchSize);
+        this.called = false;
+        this.result = null;
+    }
```

Comment:
I'd like to understand more about good validation practice for cases like this where the constructor is package private (a change I've made). If I'm the only one ever calling it, is validation necessary, or should I make certain that things are correct before they hit the Future? Or both?
Since it's only ever constructed from Computer, does having correct inputs to Computer imply correct inputs to Future? I'd think so but I could be missing something.

Reply:
Depends on how risky you want to live.  You need to assess the answer over the lifetime of the code and not just at this instant.  If you come back to the code and modify it in 2 years, will you remember the details of the contract?  What if someone else tries to modify the code?  What if someone else makes it public?

If it is impossible for a user to get to the code, you can be more loose in what you do.  Then you just need to decide what you think about devs making changes.  If you are ok with your assessment, then you can be more loose still.